### ISSUE TITLE
RaftHandle: new class to interact with RAFT

### ISSUE THREAD

**Issue by belaban** [belaban/jgroups-raft#33]
Currently, applications have to retrieve a ref to the `RAFT` and `CLIENT` protocols, which is cumbersome, as users should not have to deal with the protocol stack of JGroups.
Goal: provide a new class `RaftHandle` which sits on top of a channel and has methods such as
* set() (implementing `Settable`
* register the state machine
* get the last-applied, commit-index etc
* set `raft-id`
* Register `RoleChange` listeners

In a nutshell: let this new class handle protocol stack interactions and shield the users from the stack. 
All blocks (e.g. `CounterService` and `ReplicatedStateMachine`) should also use this class.

This class could be created on regular channels or also on fork channels.

The name is yet TBD.

**Comment by belaban** [comment]
Done, plus repackiging. `RaftHandle` is it, for now, but I'll change it when I come across a better name... :-)
